toxics/limit_data_test.go

```diff
@@ -0,0 +1,136 @@
+package toxics_test
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/Shopify/toxiproxy/stream"
+	"github.com/Shopify/toxiproxy/toxics"
+)
+
+func cmpBuffers(bufa []byte, bufb []byte) bool {
+	if len(bufa) != len(bufb) {
+		return false
+	}
+
+	for i, a := range bufa {
+		if a != bufb[i] {
+			return false
+		}
+	}
+
+	return true
+}
+
+func buffer(size int) []byte {
+	buf := make([]byte, size)
+
+	for i := 0; i < size; i++ {
+		buf[i] = byte(rand.Int())
+	}
+
+	return buf
+}
+
+func check(t *testing.T, toxic *toxics.LimitDataToxic, chunks [][]byte, expectedChunks [][]byte) {
+	input := make(chan *stream.StreamChunk)
+	output := make(chan *stream.StreamChunk, 100)
+	stub := toxics.NewToxicStub(input, output)
+
+	go func() {
+		toxic.Pipe(stub)
+	}()
+
+	for _, buf := range chunks {
+		input <- &stream.StreamChunk{Data: buf}
+	}
+
+	for _, expected := range expectedChunks {
+		chunk := <-output
+
+		if !cmpBuffers(chunk.Data, expected) {
+			t.Fail()
```

Comment:
Using `t.Error("message")` would make any test failures much more descriptive.
Something like "Chunk data did not match expected" and "Too many chunks returned by toxic"